pkg/quota/image/imagestream_evaluator.go

```diff
@@ -1,38 +1,29 @@
 package image
 
 import (
-	metainternal "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apiserver/pkg/admission"
-	kapi "k8s.io/kubernetes/pkg/api"
 	kquota "k8s.io/kubernetes/pkg/quota"
 	"k8s.io/kubernetes/pkg/quota/generic"
 
 	oscache "github.com/openshift/origin/pkg/client/cache"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
-const imageStreamEvaluatorName = "Evaluator.ImageStream"
-
 // NewImageStreamEvaluator computes resource usage of ImageStreams. Instantiating this is necessary for
 // resource quota admission controller to properly work on image stream related objects.
 func NewImageStreamEvaluator(store *oscache.StoreToImageStreamLister) kquota.Evaluator {
-	allResources := []kapi.ResourceName{
-		imageapi.ResourceImageStreams,
-	}
-
-	return &generic.GenericEvaluator{
-		Name:              imageStreamEvaluatorName,
-		InternalGroupKind: imageapi.LegacyKind("ImageStream"),
-		InternalOperationResources: map[admission.Operation][]kapi.ResourceName{
-			admission.Create: allResources,
-		},
-		MatchedResourceNames: allResources,
-		MatchesScopeFunc:     generic.MatchesNoScopeFunc,
-		ConstraintsFunc:      generic.ObjectCountConstraintsFunc(imageapi.ResourceImageStreams),
-		UsageFunc:            generic.ObjectCountUsageFunc(imageapi.ResourceImageStreams),
-		ListFuncByNamespace: func(namespace string, options metainternal.ListOptions) ([]runtime.Object, error) {
-			list, err := store.ImageStreams(namespace).List(options.LabelSelector)
+	return &generic.ObjectCountEvaluator{
+		AllowCreateOnUpdate: false,
+		InternalGroupKind:   imageapi.Kind("ImageStream"),
```

Comment:
I went through this before. I was confused initially, thinking we needed to handle both legacy and the new group. But from what I remember, InternalGroupKind is only used in 1 particular flow (I think if you're using a quota registry), and we weren't doing that, so it was ok.